peers/app_request_network.go

```diff
@@ -50,21 +54,21 @@ type AppRequestNetwork interface {
 		subnetID ids.ID,
 		allower subnets.Allower,
 	) set.Set[ids.NodeID]
+	Shutdown()
+	TrackSubnet(subnetID ids.ID)
 }
 
 type appRequestNetwork struct {
 	network         network.Network
 	handler         *RelayerExternalHandler
 	infoAPI         *InfoAPI
 	logger          logging.Logger
-	lock            *sync.Mutex
+	lock            *sync.RWMutex
 	validatorClient *validators.CanonicalValidatorClient
 	metrics         *AppRequestNetworkMetrics
 
-	// Nodes that we should connect to that are not publicly discoverable.
-	// Should only be used for local or custom blockchains where validators are not
-	// publicly discoverable by primary network nodes.
-	manuallyTrackedPeers []info.Peer
```

Comment:
👍 I like removing this from the state while keeping the functionality on `NewNetwork` initialization.